### ISSUE TITLE
HTML report for custom test tasks is generated

### ISSUE THREAD

**Issue by big-guy** [gradle/gradle#31053]
### Expected Behavior

Allow a plugin to generate an HTML report for a custom test task, by taking the binary results from the task and wiring this as input into an instance of the existing test report task.

### Context

Depends on #31051

In this milestone, the metadata is not included in the report.

No changes to improve the usability of generating the report using a separate task in this milestone.
